src/issuance.ts

```diff
@@ -47,8 +52,57 @@ export async function getIssuerUrl(issuerName: string): Promise<string> {
     }
 }
 
-export interface TokenRequestProtocol {
-    serialize(): Uint8Array;
+// Send TokenRequest to Issuer (fetch w/POST).
+export async function sendTokenRequest(
+    tokReqBytes: Uint8Array,
+    issuerUrl: string,
+    headers?: Headers,
+): Promise<{ tokResBytes: Uint8Array }> {
+    headers ??= new Headers();
+    headers.append('Content-Type', MediaType.PRIVATE_TOKEN_REQUEST);
+    headers.append('Accept', MediaType.PRIVATE_TOKEN_RESPONSE);
+    const issuerResponse = await fetch(issuerUrl, {
+        method: 'POST',
+        headers,
+        body: tokReqBytes,
+    });
+
+    if (issuerResponse.status !== 200) {
+        const body = await issuerResponse.text();
+        throw new Error(`tokenRequest failed with code:${issuerResponse.status} response:${body}`);
+    }
+
+    const contentType = issuerResponse.headers.get('Content-Type');
+
+    if (!contentType || contentType.toLowerCase() !== MediaType.PRIVATE_TOKEN_RESPONSE) {
+        throw new Error(`tokenRequest: missing ${MediaType.PRIVATE_TOKEN_RESPONSE} header`);
+    }
+
+    // Receive a stream of bytes corresponding to a serialized TokenResponse,
+    const tokResBytes = new Uint8Array(await issuerResponse.arrayBuffer());
+    return { tokResBytes };
+}
+
+export async function issuanceProtocolPub(
```

Comment:
```suggestion
export async function fetchPublicToken(
```

I could not understand the function name easily. The method could either be wrap them in one object `issuanceProtocol = { fetchPublic, fetchPrivate }` , define them as method on `TokenRequest` for each token type (might be worth having an interface), or have another name that's easier to understand.

In addition, there is a lot of similarities with `issuanceProtocolPriv`. This method shares the same issue with naming, and both might be abstracted somehow to avoid duplication, as it's only going to grow with new token types (RL, batched, grease)

Reply:
I create an interface called `PrivacyPassClient` not the best name, but allows us to move forward.